client-tez/src/test/java/org/apache/tez/runtime/library/common/shuffle/impl/RssSimpleFetchedInputAllocatorTest.java

```diff
@@ -47,6 +46,7 @@ public void testAllocate() throws IOException {
     float bufferPercent = 0.1f;
     conf.setFloat(TezRuntimeConfiguration.TEZ_RUNTIME_SHUFFLE_FETCH_BUFFER_PERCENT, bufferPercent);
     conf.setFloat(TezRuntimeConfiguration.TEZ_RUNTIME_SHUFFLE_MEMORY_LIMIT_PERCENT, 1.0f);
+    String localDirs = "/tmp/" + this.getClass().getName();
```

Comment:
Could you use `TemporaryFolder` here? Because `TemporaryFolder` will be deleted after test failed or finished.

Reply:
create folder is in method implements, not in test case.

Reply:
I know but you should have temporary directory, don't you? If we need a temporary directory, we usually use `TemporaryFolder` instead of `/tmp`. Because we don't have authority of `/tmp` sometimes and we don't clean the directory `/tmp` after the test finished, it also cause some tolerance for other test cases or other program.

Reply:
I mocked the constructor and it no need to use TemporaryFolder.

Reply:
It just ned to judge object type and no need to write local file

Reply:
OK.